src/main/java/javaslang/control/Option.java

```diff
@@ -46,6 +46,22 @@
     }
 
     /**
+     * Reduces many {@code Option}s into a single {@code Option} by transforming an
+     * {@code Iterable<Option<? extends T>>} into a {@code Option<Seq<T>>}. If any of
+     * the Options are {@link Option.None}, then this returns {@link Option.None}.
+     *
+     * @param values An {@code Iterable} of {@code Option}s.
+     * @param <T> type of the Options.
+     * @return An {@code Option} of a {@link Seq} of results.
+     */
+    static <T> Option<Seq<T>> sequence(Iterable<? extends Option<? extends T>> values) {
+        final Option<Seq<T>> zero = Option.of(List.empty());
+        final BiFunction<Option<Seq<T>>, Option<? extends T>, Option<Seq<T>>> f =
+                (result, option) -> result.flatMap(sequence -> option.map(sequence::append));
```

Comment:
It depends on the collection type. The Seq used in Future.sequence is a Stream, which has a lazy append operation taking O(1). It works like a caching iterator. Here we use List as Seq. It has O(1) prepend and O(n) append.

You find a good overview on the bottom of [this page](http://javaslang.com/javadoc/snapshot/javaslang/collection/package-summary.html).